From 6c3ac20477a4bac643088f24df3c042e627fafa9 Mon Sep 17 00:00:00 2001
From: Guillaume Burel <guillaume.burel@stormshield.eu>
Date: Fri, 3 Jan 2020 11:25:54 +0100
Subject: [PATCH] Using native binaries

Upstream-Status: Inappropriate [embedded specific]

Originally added in:
https://git.openembedded.org/meta-openembedded/commit/?id=1c8e4a679ae382f953b2e5c7a4966a4646314f3e
later extended and renamed in:
https://git.openembedded.org/meta-openembedded/commit/?id=feeb172d1a8bf010490d22b8df9448b20d9d2aed

Rebased on 24.19.0: upstream added its own '<@(emulator)' gyp variable
(defaults to empty in common.gypi) to a couple of the same action arrays this
patch touches; this recipe never sets GYP_DEFINES emulator=..., so it stays
empty and our v8-qemu-wrapper.sh insertion is unaffected either way.

Signed-off-by: Archana Polampalli <archana.polampalli@windriver.com>
---
 node.gyp                 | 3 +++
 tools/v8_gypfiles/v8.gyp | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/node.gyp b/node.gyp
--- a/node.gyp
+++ b/node.gyp
@@ -993,6 +993,7 @@
           'action_name': 'node_js2c',
           'process_outputs_as_sources': 1,
           'inputs': [
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(node_js2c_exec)',
             '<@(library_files)',
             '<@(deps_files)',
@@ -1013,6 +1014,7 @@
           ],
           'action': [
             '<@(emulator)',
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(node_js2c_exec)',
             '<@(_outputs)',
             'lib',
@@ -1072,6 +1074,7 @@
                   'action_name': 'node_mksnapshot',
                   'process_outputs_as_sources': 1,
                   'inputs': [
+                    '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
                     '<(node_mksnapshot_exec)',
                     '<(node_snapshot_main)',
                   ],
diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -113,6 +113,7 @@
         {
           'action_name': 'run_torque_action',
           'inputs': [  # Order matters.
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)torque<(EXECUTABLE_SUFFIX)',
             '<@(torque_files)',
           ],
@@ -145,6 +146,7 @@
           ],
           'action': [
             '<@(emulator)',
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)torque<(EXECUTABLE_SUFFIX)',
             '-o', '<(SHARED_INTERMEDIATE_DIR)/torque-generated',
             '-v8-root', '<(V8_ROOT)',
@@ -257,6 +259,7 @@
         {
           'action_name': 'generate_bytecode_builtins_list_action',
           'inputs': [
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)bytecode_builtins_list_generator<(EXECUTABLE_SUFFIX)',
           ],
           'outputs': [
@@ -443,6 +446,7 @@
             ],
           },
           'inputs': [
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(mksnapshot_exec)',
           ],
           'outputs': [
@@ -1976,6 +1980,7 @@
         {
           'action_name': 'run_gen-regexp-special-case_action',
           'inputs': [
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)gen-regexp-special-case<(EXECUTABLE_SUFFIX)',
           ],
           'outputs': [
--
2.34.1
